test/token/ERC20/extensions/ERC4626.test.js

```diff
@@ -44,6 +46,49 @@ contract('ERC4626', function (accounts) {
     }
   });
 
+  it('reenters deposit without changing the price', async function () {
+    const amount = web3.utils.toBN(10).pow(decimals);
+    const token = await ERC4626DepositReentrantAsset.new();
+
+    const vault = await ERC4626.new('', '', token.address);
+
+    await token.$_mint(holder, amount);
+    await token.$_mint(token.address, amount); // Requires balance to reenter
+
+    await token.approve(vault.address, constants.MAX_UINT256, { from: holder });
+
+    await token.setReenter(true);
+
+    const sharesBefore = await vault.previewDeposit(amount, { from: holder });
+    await vault.deposit(amount, holder, { from: holder });
+    const sharesAfter = await vault.previewDeposit(amount, { from: holder });
```

Comment:
It's not really clear to me why the price would be changed after the deposit is done. I think the key thing is that the attacker should not be able to get/observe a low price _during_ the reentrancy, and that's something we could test more directly by looking at the price via `convertToAssets` or the preview functions.

Reply:
The reason why the price changes is because during reentrancy there's a different proportion between assets and shares after the first `safeTransferFrom` or `mint`, which is used for the next deposit/withdraw.

The problem I see is that _during_ the reentrancy, the price is the same. You need to reenter `deposit` at least once so that you use the same price twice. Just observing the price during reentrancy is not enough

Reply:
I added extra comments so I'd say now is pretty clear since you commented on Slack that we might want to just add comments of how it works.,